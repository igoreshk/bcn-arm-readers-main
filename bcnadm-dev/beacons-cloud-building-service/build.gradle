plugins {
    id 'beacons.common-service-conventions'
}

dependencies {
    implementation('org.springframework.cloud:spring-cloud-starter-oauth2')
    implementation('org.springframework.amqp:spring-rabbit')
    implementation('org.apache.commons:commons-math3:3.6.1')
    implementation('org.geotools:gt-main:26.1')
}

jacocoTestCoverageVerification {
    violationRules {
        rule {
            limit {
                minimum = 0.80
            }
        }
    }
}
