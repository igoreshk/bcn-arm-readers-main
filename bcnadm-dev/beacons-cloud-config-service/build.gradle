plugins {
    id 'beacons.service-conventions'
}

dependencies {
    implementation('org.springframework.cloud:spring-cloud-config-server')
    implementation group: 'org.springframework.cloud', name: 'spring-cloud-starter-netflix-eureka-client'
}
