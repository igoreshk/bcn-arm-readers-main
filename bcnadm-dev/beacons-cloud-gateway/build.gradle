plugins {
    id 'beacons.service-conventions'
}

dependencies {
    implementation('org.springframework.cloud:spring-cloud-starter-config')
    implementation('org.springframework.cloud:spring-cloud-starter-netflix-eureka-client')
    implementation('org.springframework.cloud:spring-cloud-starter-openfeign')
    implementation('org.springframework.boot:spring-boot-starter-web') {
        exclude group: 'org.springframework.boot', module: 'spring-boot-starter-tomcat'
    }

    implementation group: 'org.springframework.cloud', name: 'spring-cloud-starter-oauth2'
    implementation group: 'org.springframework.cloud', name: 'spring-cloud-starter-netflix-zuul'
    implementation('org.springframework.cloud:spring-cloud-starter-security')

    compile "javax.servlet:javax.servlet-api:4.0.0"

    testImplementation('org.springframework.boot:spring-boot-starter-test')
    testImplementation('org.springframework.security:spring-security-test')
}

jacocoTestCoverageVerification {
    violationRules {
        rule {
            limit {
                minimum = 0.09
            }
        }
    }
}
