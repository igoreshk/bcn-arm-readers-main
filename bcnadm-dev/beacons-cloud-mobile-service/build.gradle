plugins {
    id 'beacons.common-service-conventions'
}

dependencies {
    implementation('org.springframework.cloud:spring-cloud-starter-oauth2')
    implementation('org.springframework.kafka:spring-kafka:2.5.1.RELEASE')
}

jacocoTestCoverageVerification {
    violationRules {
        rule {
            limit {
                minimum = 0.5
            }
        }
    }
}
