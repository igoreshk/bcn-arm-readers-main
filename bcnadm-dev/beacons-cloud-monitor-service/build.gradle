plugins {
    id 'beacons.common-service-conventions'
}

dependencies {
    implementation('org.springframework.cloud:spring-cloud-starter-oauth2')
    implementation('org.apache.commons:commons-math3:3.6.1')

    implementation('org.springframework.kafka:spring-kafka:2.5.1.RELEASE')
    implementation('org.apache.curator:curator-client:5.0.0') {
        exclude group: 'org.apache.zookeeper', module: 'zookeeper'
    }
    implementation('org.apache.curator:curator-x-async:5.0.0') {
        exclude group: 'org.apache.curator', module: 'curator-framework'
        exclude group: 'org.apache.zookeeper', module: 'zookeeper'
    }
    implementation group: 'org.apache.curator', name: 'curator-framework', version: "5.0.0"
    implementation group: 'org.apache.zookeeper', name: 'zookeeper', version: '3.6.1'
    implementation group: 'com.google.code.gson', name: 'gson', version: '2.8.6'

    implementation(group: 'io.dropwizard.metrics', name: 'metrics-core', version: '4.1.9')
    implementation(group: 'io.dropwizard.metrics', name: 'metrics-graphite', version: '4.1.9')

    implementation('org.geotools:gt-main:26.1')

    testImplementation('org.springframework.kafka:spring-kafka-test:2.5.1.RELEASE')
    testImplementation group: 'org.assertj', name: 'assertj-core', version: '3.16.1'
}

jacocoTestCoverageVerification {
    violationRules {
        rule {
            limit {
                minimum = 0.74
            }
        }
    }
}
