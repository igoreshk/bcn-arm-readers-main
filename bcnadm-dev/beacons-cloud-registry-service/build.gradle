plugins {
    id 'beacons.service-conventions'
}

dependencies {
    implementation('org.springframework.cloud:spring-cloud-starter-config')
    implementation('org.springframework.cloud:spring-cloud-starter-eureka-server:1.4.7.RELEASE') {
        exclude group: 'org.springframework.boot', module: 'spring-boot-starter-tomcat'
    }
}
