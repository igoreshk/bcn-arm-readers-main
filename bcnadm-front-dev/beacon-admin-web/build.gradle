plugins {
    id "com.github.node-gradle.node" version "2.2.4"
}

node {
    version = '12.13.0'
    download = true
    npmVersion = '6.12.0'
}

task installDependencies(type: NpmTask) {
    args = ['install']
}

task checkStyleSheets(type: NpmTask) {
    args = ['run', 'stylelint']
}
checkStyleSheets.dependsOn(installDependencies)

task checkMain(type: NpmTask) {
    args = ['run', 'lint-main']
}
checkMain.dependsOn(checkStyleSheets)

task checkTest(type: NpmTask) {
    args = ['run', 'lint-test']
}
checkTest.dependsOn(checkMain)

task test(type: NpmTask) {
    args = ['test']
}
test.dependsOn(checkTest)

// build whole frontend and
task build(type: NpmTask) {
    args = ['run', 'build']
}
build.dependsOn(test)

task start(type: NpmTask) {
    args = ['start']
}

task clean() {
    delete 'dist'
    delete 'build'
    delete 'out'
}
