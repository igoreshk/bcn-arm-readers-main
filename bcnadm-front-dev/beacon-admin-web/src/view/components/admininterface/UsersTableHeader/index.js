import React from 'react';
import { TableRow, TableCell, Button } from '@material-ui/core';
import Filter from '@material-ui/icons/FilterList';
import SortedAsc from '@material-ui/icons/KeyboardArrowDown';
import SortedDesc from '@material-ui/icons/KeyboardArrowUp';
import PropTypes from 'prop-types';

import { SORTED } from 'src/utils/usersSort';

const Unsorted = () => <SortedAsc style={{ fill: '#CCCCCC' }} />;

const headerStyle = {
  cursor: 'pointer',
  display: 'flex',
  alignItems: 'center'
};

const getIconByField = (sortedBy, field) => {
  const insideTernary = Object.is(sortedBy, SORTED[field].desc) ? <SortedDesc /> : <Unsorted />;
  return Object.is(sortedBy, SORTED[field].asc) ? <SortedAsc /> : insideTernary;
};

export const UsersTableHeader = (props) => {
  return (
    <TableRow>
      <TableCell>
        <div
          onClick={() => {
            props.toggleSortUsersByField('firstName');
          }}
          style={headerStyle}
        >
          {props.translate('admin.name')}
          {getIconByField(props.sortedBy, 'firstName')}
        </div>
      </TableCell>
      <TableCell>
        <div
          onClick={() => {
            props.toggleSortUsersByField('lastName');
          }}
          style={headerStyle}
        >
          {props.translate('admin.surname')}
          {getIconByField(props.sortedBy, 'lastName')}
        </div>
      </TableCell>
      <TableCell>
        <div
          onClick={() => {
            props.toggleSortUsersByField('email');
          }}
          style={headerStyle}
        >
          {props.translate('admin.email')}
          {getIconByField(props.sortedBy, 'email')}
        </div>
      </TableCell>
      <TableCell>{props.translate('admin.lastDate')}</TableCell>
      <TableCell>{props.translate('admin.permissions')}</TableCell>
      <TableCell>{props.translate('admin.status')}</TableCell>
      <TableCell>
        <Button style={{ float: 'right' }} onClick={props.toggleFilter}>
          <Filter />
          <span style={{ marginLeft: '8px' }}>{props.translate('admin.filter')}</span>
        </Button>
      </TableCell>
    </TableRow>
  );
};

UsersTableHeader.propTypes = {
  translate: PropTypes.func.isRequired,
  sortedBy: PropTypes.string,
  toggleFilter: PropTypes.func.isRequired,
  toggleSortUsersByField: PropTypes.func.isRequired
};
