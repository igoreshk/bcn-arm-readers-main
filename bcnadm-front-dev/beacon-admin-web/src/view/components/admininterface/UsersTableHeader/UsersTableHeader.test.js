import { UsersTableHeader } from 'src/view/components/admininterface/UsersTableHeader/index';
import React from 'react';
import { shallow } from 'enzyme/build/index';

const props = {
  translate: jest.fn(),
  toggleFilter: jest.fn(),
  toggleSortUsersByField: jest.fn()
};

const wrapper = shallow(<UsersTableHeader {...props} />);

describe('UsersTableHeader', () => {
  it('Check matching snapshot', () => expect(wrapper).toMatchSnapshot());
});
