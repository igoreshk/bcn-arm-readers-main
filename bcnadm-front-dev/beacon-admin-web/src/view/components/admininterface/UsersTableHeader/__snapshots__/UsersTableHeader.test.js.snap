// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersTableHeader Check matching snapshot 1`] = `
<WithStyles(ForwardRef(TableRow))>
  <WithStyles(ForwardRef(TableCell))>
    <div
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "cursor": "pointer",
          "display": "flex",
        }
      }
    >
      <Unsorted />
    </div>
  </WithStyles(ForwardRef(TableCell))>
  <WithStyles(ForwardRef(TableCell))>
    <div
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "cursor": "pointer",
          "display": "flex",
        }
      }
    >
      <Unsorted />
    </div>
  </WithStyles(ForwardRef(TableCell))>
  <WithStyles(ForwardRef(TableCell))>
    <div
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "cursor": "pointer",
          "display": "flex",
        }
      }
    >
      <Unsorted />
    </div>
  </WithStyles(ForwardRef(TableCell))>
  <WithStyles(ForwardRef(TableCell)) />
  <WithStyles(ForwardRef(TableCell)) />
  <WithStyles(ForwardRef(TableCell)) />
  <WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(Button))
      onClick={[MockFunction]}
      style={
        Object {
          "float": "right",
        }
      }
    >
      <Memo(ForwardRef(FilterListIcon)) />
      <span
        style={
          Object {
            "marginLeft": "8px",
          }
        }
      />
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(TableCell))>
</WithStyles(ForwardRef(TableRow))>
`;
